src/main/java/de/abstractolotl/azplace/endpoints/AuthController.java

```diff
@@ -0,0 +1,140 @@
+package de.abstractolotl.azplace.endpoints;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.server.ResponseStatusException;
+
+import de.abstractolotl.azplace.model.Session;
+import de.abstractolotl.azplace.model.User;
+import de.abstractolotl.azplace.model.UserSession;
+import de.abstractolotl.azplace.repositories.SessionRepo;
+import de.abstractolotl.azplace.repositories.UserRepo;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+
+
+@Component
+public class AuthController implements AuthAPI {
+
+    private final String casUrl;
+    private final String redirectUrl;
+    private final int    defaultKeyValidTime;
+
+    private final UserSession userSession;
+    private final SessionRepo sessionRepo;
+    private final UserRepo    userRepo;
+
+    @Autowired
+    public AuthController(UserSession userSession,
```

Comment:
Why using an constructor and not using autowired on the variables directly?
And why is every variable final?